files/enc.rb

```diff
@@ -185,7 +185,7 @@ def quote_macs! facts
 
 def build_body(certname,filename)
   puppet_facts = parse_file(filename, true)
-  hostname     = puppet_facts['values']['fqdn'] || certname
+  hostname     = puppet_facts['values']['networking']['fqdn'] || certname
```

Comment:
I think this is safer
```suggestion
  hostname     = puppet_facts.dig('values', 'networking', 'fqdn') || certname
```